### PR TITLE
[maven] add tests for bnd-maven-plugin packaging goals

### LINKED ISSUE #4960: [maven] add tests for bnd-maven-plugin packaging goals

### PR DESCRIPTION
**Pull request by rotty3000**
fixes #4960

Signed-off-by: Raymond Augé <raymond.auge@liferay.com>

### CODE REVIEW

**Comment by rotty3000** [comment]
There's an absolutely tiny flaw in the _wrapping_ use case when using the `jar` goal but the work around is so trivial that I'm tempted to fix the bug later and not further defer the already late release process for 6.2.0

**Comment by rotty3000** [comment]
I've created https://github.com/bndtools/bnd/issues/5080 to track the issue.

**Review by bjhargrave — APPROVED** [review]
Thanks!

**Comment by rotty3000** [comment]
kk... we're all good now.

File: maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/AbstractBndMavenPlugin.java [hunk 1]
```diff
@@ -565,6 +565,15 @@ private static StringBuilder addHeaderAttribute(StringBuilder builder, String ke
 
 	private void attachArtifactToProject(Jar bndJar) {
 		File artifactFile = createArtifactFile();
+		File outputDir = artifactFile.getParentFile();
+
+		if (!outputDir.exists()) {
+			try {
+				IO.mkdirs(outputDir);
+			} catch (IOException ioe) {
+				Exceptions.duck(ioe);
```
Comment (bjhargrave) [inline maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/AbstractBndMavenPlugin.java:574]:
```suggestion
				throw Exceptions.duck(ioe);
```

We always want to use the construct "throw the duck" as this tells the compiler there is a flow control change.

**Review by bjhargrave — CHANGES_REQUESTED** [review]

File: maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/AbstractBndMavenPlugin.java (hunk 1 above)
Comment (rotty3000) [reply maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/AbstractBndMavenPlugin.java]: good catch, I missed that one. fixed!

**Review by bjhargrave — APPROVED** [review]

**Pull request merged**